Documented Http shards plus others

ShahFragcolor opened the pull request:
- Documented all Http shards
- documented ForEach and Map


<!--
Before submitting your PR, please review the following checklist:

- [ ] **DO NOT** submit a PR without having discussed the change first in a related issue. If none exist, create one first.
- [ ] **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] **DO** keep pull requests small so they can be easily reviewed.
- [ ] **DO** make sure all unit tests pass.
- [ ] **DO** make sure not to introduce any new compiler warnings.
- [ ] **AVOID** breaking the continuous integration build.
- [ ] **AVOID** making significant changes to the overall architecture.
-->

sinkingsugar commented on shards/modules/core/core.cpp:
```diff
@@ -889,6 +896,18 @@ struct ForEachShard {
 };
 
 struct Map {
+  static SHOptionalString help() {
```
I think we wanna do a quick mention of $0 and $1 here as well, short.

sinkingsugar commented on shards/modules/core/core.cpp:
```diff
@@ -2687,6 +2729,20 @@ struct PassShard : public LambdaShard<unreachableActivation, CoreInfo::AnyType,
 
+struct HasherShard : public LambdaShard<hashActivation, CoreInfo::AnyType, CoreInfo::Int2Type> {
+  static SHOptionalString help() {
+    return SHCCSTR("This shard takes any input type, hashes them using the XXH128 hashing algorithm and returns their 128-bit hash value as a a sequence with 2 64-bit integer as elements.");
```
it's not a sequence, but just an Int2 to be correct

sinkingsugar reviewed (approved):
Looks good, just one minor change to do

sinkingsugar commented on shards/modules/crypto/src/hash.rs:
```diff
@@ -149,6 +151,39 @@ macro_rules! add_hasher {
+
+      fn help(&mut self) -> OptionalString {
+        if $name_str == "Hash.Keccak-256" {
+          OptionalString(shccstr!("This shard converts the input (string, bytes or a sequence of strings/bytes) into a 256-bit hash(of 32 bytes) using the Keccak-256 hashing algorithm."))
```
hmm , it does not really "convert", just outputs the hash of it

sinkingsugar commented on shards/modules/crypto/src/hash.rs:
```diff
@@ -149,6 +151,39 @@ macro_rules! add_hasher {
+
+      fn outputHelp(&mut self) -> OptionalString {
+        if $name_str == "Hash.Keccak-256" {
+          OptionalString(shccstr!("Returns the 256-bit hash as a byte sequence (returns a sequence of bytes instead if the input was a sequence)."))
```
Something I forgot to mention that might be everywhere in our codebase sadly.. In Shards there is no real return, but rather output :)

sinkingsugar commented on shards/modules/json/json.cpp:
```diff
@@ -617,6 +617,14 @@ struct ToJson {
+  static SHOptionalString help() { return SHCCSTR("This shard takes its input and converts it into a JSON string."); }
+
+  static SHOptionalString inputHelp() {
+    return SHCCSTR("If Pure is set to true, this shard acceps an input of anytype. If Pure is set to false, this shard only "
```
This is wrong, pure stands for `pure` json, sort of compatible with JS, you should check the output, non pure is instead shards flavored json, (with shards types info)

sinkingsugar commented:
@ShahFragcolor this is still pending for fixes

ShahFragcolor committed:
Made requested changes

- changed Hash shards help text from "converts the input" to "outputs the hash of the input ..."
- Changed ToJson input help text to be more clear
